File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start <=0) return ERROR_VALUE;

    int numSteps = 0;
    int value = start;
    
    while (value > 1) {
        if (value % 2 ==0) {
            value = value / 2;
        } else {
            value = (3 * value) + 1;
        }

        numSteps++;
    }
    
    return numSteps;
}

File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number <= 0)
            throw new ArgumentOutOfRangeException();
        
        var numberOfSteps = 0;

        while (number != 1)
        {
            number = number % 2 == 0
                ? number / 2
                : (number * 3) + 1;
                
            numberOfSteps++;
        }
        
        return numberOfSteps;
    }
}